### PR TITLE
Clear cache on delete deployment

### PR DESCRIPTION
**Pull request by pramodcs**

### CODE REVIEW

File: ushahidi/src/main/java/com/ushahidi/android/domain/usecase/deployment/DeleteDeploymentUsecase.java [hunk 1]
```diff
@@ -48,6 +73,11 @@ protected Observable buildUseCaseObservable() {
         if (mDeploymentId == null) {
             throw new RuntimeException("Deployment ID is null. You must call setDeployment(...)");
         }
+        mPostRepository.deleteDeploymentPosts(mDeploymentId);
```
Comment (eyedol) [inline ushahidi/src/main/java/com/ushahidi/android/domain/usecase/deployment/DeleteDeploymentUsecase.java]: You should make individual use cases for deleting a particular entity. There should be use case for deleteGeoJsonUsecase, deleteFormUsecase, etc, Then call them up in the presenter. You've already done most of the hard work

**Comment by eyedol** [comment]
@pramodcs  any progress on this?

**Comment by pramodcs** [comment]
I couldn't spend time on this last week. Why does the use case needs to be broken down to that level. The only time all the deployment related data is deleted is when the deployment is deleted.

**Comment by eyedol** [comment]
Because your current implementation is manipulating data related to other entities. Makes reading the code more confusing from the domain layer.

**Comment by pramodcs** [comment]
Could you elaborate on how you would like the Presenter to implement the chain of deleteUseCase calls?

**Comment by eyedol** [comment]
@pramodcs This looks good. Thanks for taking the trouble to break down the usecase. I see that the `PR` is conflicting with the master branch. Could you fix the merge conflicts so I can merge this? Thanks.

**Comment by pramodcs** [comment]
Yep. I was waiting to see if any other changes are needed. I will resolve the conflicts.

**Comment by pramodcs** [comment]
@eyedol You can review again.

**Comment by eyedol** [comment]
@pramodcs  Looks good to me :+1:

**Pull request merged**